### ISSUE TITLE
Question regarding relative urls in importmap scopes

### ISSUE THREAD

**Issue by dmail** [guybedford/es-module-shims#90]
Hi there, 

Inside an importmap file I would expect relative urls found in `scopes` to be resolved against the importmap file url.
According to my test, they are resolved against the importer file.

## Steps to reproduce

- Create an empty directory

- Create `dist/bare.js` file

```js
```

- Create `dist/index.js` file
```js
import "bare"
```

- Create `importmap.importmap` file

```json
{
  "scopes": {
    "./dist/": {
      "bare": "./dist/bare.js"
    }
  }
}
```

- Create `index.html` file
```html
<html>
  <head>
    <title>Test</title>
    <meta charset="UTF-8" />
  </head>

  <body>
    <script defer src="https://unpkg.com/es-module-shims@0.2.13/dist/es-module-shims.js"></script>
    <script type="importmap-shim" src="./importmap.importmap"></script>
    <script type="module-shim" src="./dist/index.js"></script>
  </body>
</html>
```

- Your file structure looks like this

```txt
dist/
  bare.js
  index.js
importmap.importmap
index.html
```

- Start an http server and open `index.html`

## Actual

An error is displayed in the console saying

```
Uncaught (in promise) Error: 404 Not Found http://127.0.0.1:8080/dist/dist/bare.js
    at es-module-shims.js:893
```

## Expected

No error in the console

## Notes

I expected `"./dist/bare.js"` to be resolved to `http://127.0.0.1:8080/dist/bare.js` but it is resolved to `http://127.0.0.1:8080/dist/dist/bare.js` (`dist` appear twice).

When I look at reference implementation of importmaps where importmap scopes are normalized I understand that relative url in scopes are resolved against the importmap file base url (which fallbacks to document url for inline importmap).
See https://github.com/WICG/import-maps/blob/a3e697d7b29d929e099cc012768245f8fd07597d/reference-implementation/lib/parser.js#L99

Can you confirm it works as designed and that you are spec compliant ?

**Comment by guybedford** [comment]
@dmail yes these cases are supposed to resolve against the import map file path itself, so if that is not the case then this is a bug.

The resolveAndComposeImportMap function call here - https://github.com/guybedford/es-module-shims/blob/master/src/es-module-shims.js#L231 is supposed to carry through the import map URL as the base URL for resolutions when the map is resolved on load. As far as I can tell looking at the code that is what it is doing.

If you want to look into this further or post a PR please do.

**Comment by guybedford** [comment]
I just released 0.6.0 with the fix.

**Comment by dmail** [comment]
Just tested with version 0.6.0 and it works like a charm. Thank you :)
